File: front/src/app/booking/booking.ts
import { Guid } from "guid-typescript";
import { Purpose } from "../purpose/purpose";
import { Room } from "../room/room";

export interface Booking {
    id?: Guid,
    userId: number,
    purposeId: number,
    roomId: number,
    dateIn: Date,
    dateOut: Date,
    purpose?:Purpose
    room?:Room,
    participants?: string,
    array?: string[]
}

File: back/Models/Booking.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace workorooms.Models
{
    public class Booking
    {
        [Key]
        public Guid Id { get; set; }
        public int UserId { get; set; }
        public int PurposeId { get; set; }
        public int RoomId { get; set; }
        public DateTime DateIn { get; set; }
        public DateTime DateOut { get; set; }
        public User User { get; set; }
        public Purpose Purpose { get; set; }
        public Room Room { get; set; }
        public string Participants { get; set; }
    }
}
